src/mlpack/methods/preprocess/preprocess_string_util.hpp

```diff
@@ -0,0 +1,71 @@
+/**
+ * @file preprocess_string_util.cpp
+ * @author Jeffin Sam
+ *
+ * A CLI executable to encode string dataset.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+#ifndef MLPACK_CORE_DATA_STRING_UTIL_HPP
+#define MLPACK_CORE_DATA_STRING_UTIL_HPP
+
+#include <mlpack/prereqs.hpp>
+#include <mlpack/core/util/mlpack_main.hpp>
```

Comment:
`mlpack/core/util/mlpack_main.hpp` should only be included in CLI main files. Remove the line, it should fix the error. Also remove redundant includes.